lib/SILOptimizer/SemanticARC/OwnershipLiveRange.cpp

```diff
@@ -278,6 +278,7 @@ static SILValue convertIntroducerToGuaranteed(OwnedValueIntroducer introducer) {
   case OwnedValueIntroducerKind::Phi: {
     auto *phiArg = cast<SILPhiArgument>(introducer.value);
     phiArg->setOwnershipKind(OwnershipKind::Guaranteed);
+    phiArg->setReborrow(computeIsReborrow(phiArg));
```

Comment:
Would it make sense to have a method on `SILPhiArgument` like `recomputeFlags` (maybe with a bitfield argument where each bit indicates which flag and the default value is "all flags")?  Otherwise we might end up with a series of calls to `arg->setFoo(computeIsFoo(arg))`.

Reply:
Checking my understanding, that means we won't see code like
```
phiArg->setReborrow(computeIsReborrow(phiArg))
phiArg->setEscaping(computeIsEscaping(phiArg))
```
Out of curiosity, why is that unlikely?

Reply:
I don't expect transforms to introduce "escaping" at the same time they introduce "reborrow". "reborrow" flag can be introduced in any ownership / control flow transforms that move from "owned" -> "reborrow" or "guaranteed forwarding" to "reborrow". 
"escaping" flag will be computed in the mandatory pipeline and maybe introduced while inlining, or reset during dce. 

Also, I have not worked on populating "ecaping" flag so can't deduce all possible places it needs to be set/reset.  I will revisit adding this composite api if I see this becoming an issue.